packages/web/hooks/limit-orders/use-place-limit.ts

```diff
@@ -408,13 +408,23 @@ const useLimitPrice = ({
   orderDirection: OrderDirection;
   baseDenom?: string;
 }) => {
-  const { data, isLoading } = api.edge.orderbooks.getOrderbookState.useQuery({
-    osmoAddress: orderbookContractAddress,
-  });
+  const { data, isLoading } = api.edge.orderbooks.getOrderbookState.useQuery(
+    {
+      osmoAddress: orderbookContractAddress,
+    },
+    {
+      enabled: Boolean(orderbookContractAddress),
+    }
+  );
   const { data: assetPrice, isLoading: loadingAssetPrice } =
-    api.edge.assets.getAssetPrice.useQuery({
-      coinMinimalDenom: baseDenom ?? "",
-    });
+    api.edge.assets.getAssetPrice.useQuery(
+      {
+        coinMinimalDenom: baseDenom ?? "",
+      },
+      {
+        enabled: Boolean(baseDenom) && baseDenom!.length > 0,
```

Comment:
Why are you checking for the length of the `baseDenom` string?
Also, you can just use !!baseDenom without the boolean constructor